plugins/modules/aci_fabric_node_control.py

```diff
@@ -0,0 +1,268 @@
+                    "ownerTag": ""
+                }
+            }
+        }
+    ]
+proposed:
+  description: The assembled configuration from the user-provided parameters
+  returned: info
+  type: dict
+  sample:
+    {
+        "fvTenant": {
+            "attributes": {
+                "descr": "Production environment",
+                "name": "production"
+            }
+        }
+    }
+filter_string:
+  description: The filter string used for the request
+  returned: failure or debug
+  type: str
+  sample: ?rsp-prop-include=config-only
+method:
+  description: The HTTP method used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: POST
+response:
+  description: The HTTP response from the APIC
+  returned: failure or debug
+  type: str
+  sample: OK (30 bytes)
+status:
+  description: The HTTP status from the APIC
+  returned: failure or debug
+  type: int
+  sample: 200
+url:
+  description: The HTTP url used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: https://10.11.12.13/api/mo/uni/tn-production.json
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.aci.plugins.module_utils.aci import ACIModule, aci_argument_spec, aci_annotation_spec, aci_owner_spec
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(aci_annotation_spec())
+    argument_spec.update(aci_owner_spec())
+    argument_spec.update(
+        name=dict(type="str", aliases=["fabric_node_control"]),
+        description=dict(type="str"),
+        enable_dom=dict(type="bool"),
+        feature_selection=dict(type="str", choices=["analytics", "netflow", "telemetry"], aliases=["feature"]),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+    )
+    aci = ACIModule(module)
+
+    name = module.params.get("name")
+    description = module.params.get("description")
+    enable_dom = aci.boolean(module.params.get("enable_dom"), "Dom", "None")
+    feature_selection = module.params.get("feature_selection")
+    state = module.params.get("state")
+
+    aci.construct_url(
+        root_class=dict(
+            aci_class="fabricNodeControl",
+            aci_rn="fabric/nodecontrol-{0}".format(name),
+            module_object=name,
+            target_filter={"name": name},
+        ),
+    )
+    aci.get_existing()
+
+    if state == "present":
+        aci.payload(
+            aci_class="fabricNodeControl",
+            class_config=dict(
+                name=name,
+                descr=description,
+                control=enable_dom,
+                featureSel=feature_selection,
+            ),
+        )
+
+        aci.get_diff(aci_class="fabricNodeControl")
+
+        aci.post_config()
+
+    elif state == "absent":
+        aci.delete_config()
+
```

Comment:
Does `query` state need to be handled or is that done via another mechanism?

Reply:
Query state is already handled in the `aci.get_existing()`, which retrieves the current configuration. We always execute this to populate the previous value and calculate the changed state in the return.